### PR TITLE
Default dtype and device to SB environment

### PR DESCRIPTION
**Pull request by bqth29**
# 💬 Pull Request Description

The `get/set/reset_env` functions have been moved to their own file named `environment.py` and all the dtype and device occurences in methods signatures have been made optional with a default value to `None`.

# ✔️ Check list

- [x] The code matches the styling rules
- [x] The new code is covered by relevant tests
- [x] Documentation was added

# 🚀 New features

None

# 🐞 Bug fixes

None

# 📣 Supplementary information

The configure default values for the tensors' dtype and device, it is recommanded to ather use PyTorch's `torch.set_default_dtype` and `torch.set_default_device` functions.

We considered the possibility of configuring these values via the SB environment directly, but this would risk creating a mismatch between the information stored in the environment and the actual default value, should a call be made to one of PyTorch's functions in the meantime.

### CODE REVIEW

File: tests/test_environment.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        pressure_slope=0.1,
+        heat_coefficient=0.08,
+    )
+    assert {
+        "TIME_STEP": 0.05,
+        "PRESSURE_SLOPE": 0.1,
+        "HEAT_COEFFICIENT": 0.08,
+    } == get_env()
+    reset_env()
+
```
Comment (BusyBeaver-42) [inline tests/test_environment.py:31]: We should also test if `set_env` behaves properly when some of its arguments are `None`.

```suggestion

def test_set_env_none():
    set_env(pressure_slope=0.2)
    set_env(heat_coefficient=0.04)
    assert {
        "TIME_STEP": 0.1,
        "PRESSURE_SLOPE": 0.2,
        "HEAT_COEFFICIENT": 0.04,
    } == get_env()
    
```

**Review by BusyBeaver-42 — APPROVED** [review]
